app/src/main/kotlin/com/gdgnantes/devfest/android/content/RemindersHandler.kt

```diff
@@ -0,0 +1,129 @@
+package com.gdgnantes.devfest.android.content
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.os.Bundle
+import android.support.v4.app.NotificationCompat
+import android.support.v4.app.NotificationManagerCompat
+import android.support.v4.content.ContextCompat
+import android.util.Log
+import com.gdgnantes.devfest.android.BookmarkManager
+import com.gdgnantes.devfest.android.R
+import com.gdgnantes.devfest.android.SessionActivity
+import com.gdgnantes.devfest.android.database.sqlIn
+import com.gdgnantes.devfest.android.format.text.DateTimeFormatter
+import com.gdgnantes.devfest.android.model.toRoom
+import com.gdgnantes.devfest.android.model.toSession
+import com.gdgnantes.devfest.android.provider.ScheduleContract
+import java.util.*
+
+class RemindersHandler(private val context: Context) {
+
+    // NOTE Cyril
+    // The current implementation of the reminders consists on
+    // setting on alarm (in the AlarmManager) that will be triggered
+    // exactly REMINDER_ANTICIPATION milliseconds before the next
+    // (in time) bookmarked event. The alarm's PendingIntent
+    // contains all of the information of the next bookmarked events.
+    // Note that it's event*s* because, although it should be quite
+    // rare, there may be several bookmarked events at the same
+    // time.
+    //
+    // Once triggered, the alarm will display a notification on a
+    // per-event basis. In case multiple bookmarked event start at
+    // the same time, the system will hence trigger several
+    // notifications at the same time.
+
+    companion object {
+        const val CHANNEL_REMINDERS = "channel:reminders"
+
+        private const val TAG = "RemindersHandler"
+
+        private const val KEY_ID = "key:id"
+        private const val KEY_TITLE = "key:title"
+        private const val KEY_TEXT = "key:text"
+
+        private const val NOTIFICATION_ID = 666
+
+        private const val REMINDER_ANTICIPATION = 5 * 60 * 1000L // 5 minutes
+    }
+
+    fun updateAlarm() {
+        val bookmarkIds = BookmarkManager.from(context).getLiveData().value ?: emptySet()
```

Comment:
May be done in a background `Thread` but `BookmarkManager.from` in `@MainThread`